extra/test_release_tool.py

```diff
@@ -419,32 +436,38 @@ def test_get_components_of_type(integration_dir_func):
     # only_non_release=True
     repos_comp = Component.get_components_of_type("git", only_non_release=True)
     repos_name = [r.name for r in repos_comp]
-    assert "deployments" not in repos_name
     assert "deployments-enterprise" not in repos_name
     assert "deviceauth" not in repos_name
     assert "gui" not in repos_name
     assert "integration" not in repos_name
-    assert "inventory" not in repos_name
     assert "inventory-enterprise" not in repos_name
     assert "mender" not in repos_name
     assert "mender-artifact" not in repos_name
     assert "mender-cli" not in repos_name
     assert "tenantadm" not in repos_name
-    assert "useradm" not in repos_name
     assert "useradm-enterprise" not in repos_name
-    assert "workflows" not in repos_name
     assert "workflows-enterprise" not in repos_name
     assert "create-artifact-worker" not in repos_name
     assert "auditlogs" not in repos_name
     assert "mtls-ambassador" not in repos_name
     assert "deviceconnect" not in repos_name
     assert "mender-connect" not in repos_name
     assert "deviceconfig" not in repos_name
+    if not is_staging:
+        assert "deployments" not in repos_name
+        assert "inventory" not in repos_name
+        assert "useradm" not in repos_name
+        assert "workflows" not in repos_name
```

Comment:
This approach does not seem very maintainable. Isn't it better to deduce this from looking at the "-enterprise" suffix? It will only happen to services that have both OS and ENT components, right?

Reply:
> It will only happen to services that have both OS and ENT components, right?

Correct



> This approach does not seem very maintainable.

The whole test is not very maintainable, really. Adding a new repo will still make the this test pass, but won't really verify that the repo is there... (it will however break the next test, `test_list_repos`).

Maybe I should just remove these test? I added them when I implementing the `--only-backend` flag to make sure I didn't break existing functionality.

Reply:
Hmm yes, tricky... Maybe we can just test some known repositories, without testing the whole list?

I don't really have a strong preference, you're right that it already a bit unmaintainable, but I don't see another way to test this without hardcoding.

Reply:
Okey: I'll rewrite to verify _some_ repos, making sure I catch the different "types" we have. Then is obvious we are not testing all but still verifying the gist of the functionality.